libraries/bot-azure/src/test/java/com/microsoft/bot/azure/AzureQueueTests.java

```diff
@@ -0,0 +1,234 @@
+            EMULATOR_IS_RUNNING = true;
+        }
+    }
+
+    // These tests require Azure Storage Emulator v5.7
+    public QueueClient containerInit(String name) {
+        QueueClient queue = new QueueClientBuilder().endpoint(AZURE_STORAGE_EMULATOR_ENDPOINT).buildClient();
+        queue.create();
+        queue.clearMessages();
+        return queue;
+    }
+
+    @Test
+    public void continueConversationLaterTests() {
+        assertEmulator();
+        String queueName = "continueconversationlatertests";
+        QueueClient queue = containerInit(queueName);
+        ConversationReference cr = TestAdapter.createConversationReference("ContinueConversationLaterTests", "User1", "Bot");
+        TestAdapter adapter = new TestAdapter()
+            .useStorage(new MemoryStorage())
+            .useBotState(new ConversationState(new MemoryStorage()), new UserState(new MemoryStorage()));
+
+        AzureQueueStorage queueStorage = new AzureQueueStorage(connectionString, queueName);
+        ContinueConversationLater ccl = new ContinueConversationLater() {
+            {
+                setDate("=addSeconds(utcNow(), 2)");
+                setValue("foo");
+            }
+        };
+        DialogManager dm = new DialogManager(ccl, "DialogStateProperty");
+        dm.getInitialTurnState().replace("QueueStorage", queueStorage);
+
+        new TestFlow(adapter, turnContext -> {
+           return dm.onTurn(turnContext).thenAccept(null);
+        })
+            .send("hi")
+            .startTest().join();
+        try {
+            Thread.sleep(DEFAULT_DELAY);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+
+        QueueMessageItem messages = queue.receiveMessage();
+        JacksonAdapter jacksonAdapter = new JacksonAdapter();
+        String messageJson = messages.getMessageText();
+        Activity activity = null;
+        try {
+            activity = jacksonAdapter.deserialize(messageJson, Activity.class);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+        Assert.assertTrue(activity.isType(ActivityTypes.EVENT));
+        Assert.assertEquals(ActivityEventNames.CONTINUE_CONVERSATION, activity.getName());
+        Assert.assertEquals("foo", activity.getValue());
+        Assert.assertNotNull(activity.getRelatesTo());
+        ConversationReference cr2 = activity.getConversationReference();
+        cr2.setActivityId(null);
+        cr2.setActivityId(null);
+        try {
+        Assert.assertEquals(jacksonAdapter.serialize(cr), jacksonAdapter.serialize(cr2));
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+
+    }
+
+    private void assertEmulator() {
+        if (!EMULATOR_IS_RUNNING) {
+            Assert.fail(NO_EMULATOR_MESSAGE);
+        }
+    }
+
+    private class ContinueConversationLater extends Dialog {
```

Comment:
Wouldn't it be better to put this class in a new file?

Reply:
This is something of `adaptiveDialogs`, this was not migrated in `bot-dialogs`. As Tracy said, we won't have parity of adaptive-dialogs right now.. so I don't want to add noise in that library. it's used just for testing.

Reply:
Maybe during the testing we can find another approach instead of adding this class here, maybe generating another dialog